util/internal/integrationtest/mapper/orm/src/main/java/org/hibernate/search/util/impl/integrationtest/mapper/orm/OrmSetupHelper.java

```diff
@@ -189,17 +192,39 @@ protected void close(SessionFactory toClose) {
 		}
 	}
 
+	@Override
+	public void afterTestExecution(ExtensionContext context) {
+		// if test was aborted then we don't want to clean the data since the test wasn't executed.
+		if ( !context.getExecutionException().map( Object::getClass )
+				.map( org.opentest4j.TestAbortedException.class::equals ).orElse( Boolean.FALSE ) ) {
+			this.ormSetupHelperCleaner.cleanupData( sessionFactory );
+		}
```

Comment:
mm right... I've changed the order somewhat on purpose 😃. assuming we have a test that configures this helper (and builds a session factory) in a before method we can't do the cleanup earlier then this before is executed... then the helper will destroy the session factory in the corresponding after (all/each). so that's why it cleans the data after the execution and uses the same session factory that was used to run the tests....

Reply:
> assuming we have a test that configures this helper (and builds a session factory) in a before method we can't do the cleanup earlier then this before is executed

If it's a `@BeforeEach` method, then there's no point clearing data anyway, the session factory should be closed.

If it's a `@BeforeAll` method... I don't really understand the problem. You can (and probably should) run your stuff in a beforeEachCallback, no?